pos-print/src/main/java/com/clouway/pos/print/adapter/http/PrintService.kt

```diff
@@ -25,22 +26,33 @@ class PrintService @Inject constructor(private var factory: PrinterFactory) {
     try {
       val dto = request.read(ReceiptDTO::class.java).`as`(GsonTransport::class.java)
       val printer = factory.getPrinter(dto.sourceIp)
-      when {
-        dto.fiscal
-        -> response = printer.printFiscalReceipt(dto.receipt)
-        else
-        -> response = printer.printReceipt(dto.receipt)
+      try {
+        when {
+          dto.fiscal
+          -> response = printer.printFiscalReceipt(dto.receipt)
+          else
+          -> response = printer.printReceipt(dto.receipt)
+        }
+      } catch (e: RequestTimeoutException) {
+        e.printStackTrace()
+        return Reply.with(ErrorResponse("Printer request timeout.")).`as`(GsonTransport::class.java).badRequest()
```

Comment:
Doesn't look reasonable timeout to be returned as bad request. Temporary not available sounds like a better indicator that device becomes unavailable.